src/computational_graph/graph.jl

```diff
@@ -30,24 +40,27 @@ Base.show(io::IO, ::Type{Prod}) = print(io, "Ⓧ")
 
 # Example:
 ```julia-repl
-julia> g = Graph([𝑓⁺(1)𝑓⁻(2), 𝑓⁺(3)𝑓⁻(4)], external=[1, 2], subgraphs=[Graph([𝑓⁺(1)𝑓⁻(4)], []), Graph([𝑓⁻(2)𝑓⁺(3)], [])])
-3:f⁺(1)f⁻(2)|f⁺(3)f⁻(4)=0.0=⨁ (1,2)
+julia> g1 = Graph([], vertices=[𝑓⁺(1),𝑓⁻(2)], external=[1,2], hasLeg=[true,true])
+1:f⁺(1)|f⁻(2)=0.0
 
-julia> g.subgraphs
-2-element Vector{Graph{Float64, Float64}}:
- 1:f⁺(1)f⁻(4)=0.0
- 2:f⁻(2)f⁺(3)=0.0
+julia> g2 = Graph([], vertices=[𝑓⁺(3),𝑓⁻(4)], external=[1,2], hasLeg=[true,true])
+2:f⁺(3)|f⁻(4)=0.0
+
+julia> g = Graph([g1,g2], vertices=[𝑓⁺(1),𝑓⁻(2),𝑓⁺(3),𝑓⁻(4)], operator=ComputationalGraphs.Prod(), external=[1,2,3,4], hasLeg=[true,true,true,true])
+3:f⁺(1)|f⁻(2)|f⁺(3)|f⁻(4)=0.0=Ⓧ (1,2)
 ```
 """
 mutable struct Graph{F,W} # Graph
     id::Int
     name::String # "" by default
-    type::Symbol # :propagator, :interaction, :sigma, :green, :generic
+    # type::Symbol # :propagator, :interaction, :sigma, :green, :generic
+    type::DataType
```

Comment:
Why are we claim type::DataType here? We don't(and can't) do dispatch in this case(I see dispatch only in constructors and in all other cases it's type==XXX), and it's a bit weird that the constructor of Graph gets an concrete object and results in a DataType

Reply:
In other words, I didn't see advantages of current implementation comparing to the previous Symbol version